src/lib/typeUtils.ts

```diff
@@ -12,3 +12,10 @@ export function orEmptyObj<T extends object>(obj: T | null | undefined): Partial
 export function isPresent<T>(value: T | undefined | null): value is T {
   return value != null;
 }
+
+export function isValidKey<K extends keyof any>(
```

Comment:
@tyleroooo maybe we should just make a `typesafeAccess` function...? 
in fact we can maybe even do something like
```
Object.prototype.get = function(k, default = undefined) {
  return isValidkey(k, this) ? this[k] : default
};
```

although

Reply:
yeah this seems fine for now